### PR TITLE
Add sphobjinv recipe to conda-forge

### PR DESCRIPTION
**Pull request by anjos**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by anjos** [comment]
@bskinn: please confirm your willingness to become a co-maintainer of this package as per conda-forge rules.

**Comment by anjos** [comment]
@ocefpaf: this PR concludes the work started on #21238, so we'd appreciate your advice.  This is a noarch package, but it fails on Windows.  I do not understand why it is tested on windows in the first place, nor why it fails.  @bskinn - any ideas?

**Comment by anjos** [comment]
@conda-forge/help-python I'm tagging this for review.

File: recipes/sphobjinv/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+{% set name = "sphobjinv" %}
+{% set version = "2.3" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/bskinn/sphobjinv/archive/refs/tags/v{{ version }}.tar.gz
```
Comment (ocefpaf) [inline recipes/sphobjinv/meta.yaml:9]: I'm guessing they are not publishing the sdist on PyPI, right? :weary:

File: recipes/sphobjinv/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  sha256: 29c2287afe54682bdc6ff66337c36af7dda63b1198166bae7aa0185012e46dde
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
```
Comment (ocefpaf) [inline recipes/sphobjinv/meta.yaml:15]: You are missing the entry_point here. This should fix the Windows build.

```suggestion
  noarch: python
  script: {{ PYTHON }} -m pip install . -vv
  number: 0
  entry_points:
    - sphobjinv = sphobjinv.cli.core:main
```

File: recipes/sphobjinv/meta.yaml (hunk 1 above)
Comment (anjos) [reply recipes/sphobjinv/meta.yaml:9]: Right, not yet - may change in a future release.  I'm working with @bskinn on this item.

File: recipes/sphobjinv/meta.yaml (hunk 2 above)
Comment (anjos) [reply recipes/sphobjinv/meta.yaml:15]: Right, thx for catching this.  One of the issues of not using grayskull, clearly!

Comment (ocefpaf) [reply recipes/sphobjinv/meta.yaml:15]: Yes and no. Grayskull does a great job at finding these in setup.py. I'm not sure if setup.cfg is supported and I know pyporject.toml is not. We are working on it though. BTW, you can use grayskull on GitHub releases.

File: recipes/sphobjinv/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    - sphobjinv
+  source_files:
+    - README.rst
+    - tox.ini
+    - requirements-dev.txt
+    - conftest.py
+    - tests/
+  commands:
+    - pip check
+    - pytest -k 'not readme'
```
Comment (ocefpaf) [inline recipes/sphobjinv/meta.yaml:49]: This test is failing on Windows now, not sure why. But, if that is not a deal breaker we can merge this b/c the package is noarch.

Comment (bskinn) [reply recipes/sphobjinv/meta.yaml:49]: I've ID-ed the problem and am narrowing down on a fix. Will be getting a new sdist to @anjos soon, which will hopefully provide an all-green test board.

Comment (bskinn) [reply recipes/sphobjinv/meta.yaml:49]: (There's been a rabbit hole of Win vs Unix EOL silliness to dive down.)

**Comment by bskinn** [comment]
@anjos:

> @ocefpaf: this PR concludes the work started on #21238, so we'd appreciate your advice. This is a noarch package, but it fails on Windows. I do not understand why it is tested on windows in the first place, nor why it fails. @bskinn - any ideas?

It's tested on Windows because, if used on Windows, it needs to correctly handle EOLs when working with decompressed inventories. (This is less *strictly* true than it used to be, with Windows tools getting better at tolerating Unix EOLs, but is still necessary IMO.)

I'm not sure exactly which rev you're testing now, so I can't speak precisely to the Win test failures without more info. I think I've got everything chased down, the Azure Pipelines are looking good in this most recent CI run on sphobjinv#268. Not sure why the GH Actions job failed, now, though

**Comment by bskinn** [comment]
Ah. Newly broken link in my Sphinx docs. Yay.

Definitely not an issue for the conda-forge packaging. 

@anjos , new sdist coming shortly

**Comment by anjos** [comment]
> It's tested on Windows because, if used on Windows, it needs to correctly handle EOLs when working with decompressed inventories. (This is less _strictly_ true than it used to be, with Windows tools getting better at tolerating Unix EOLs, but is still necessary IMO.)

@bskinn: I meant the conda package itself not your upstream.  The so-called `noarch` conda-forge packages only build/test on linux.  I'm not sure if there is any conda-forge provision to generate a single noarch package, but still test it across various architectures.  @ocefpaf may be able to confirm this.

**Comment by bskinn** [comment]
> bskinn: I meant the conda package itself, not your upstream. 

Aha, that's what I get for trying to diagnose on my phone.

Change the pytest invocation to use double-quotes around `not readme`. Single quotes apparently don't mean "treat this as a single argument" to Windows... sigh.

**Comment by bskinn** [comment]
Full disclosure: `sphobjinv` ***does*** currently vendor an old version of `fuzzywuzzy`, tweaked so that the pieces that `sphobjinv` needs are Python 3 compatible. 

This old version is required due to a licensing conflict with newer `fuzzywuzzy`. Simply pinning to this license-compatible old version is not feasible because of the lack of Python 3 compatibility available at that version. Vendoring and adapting seemed the least-worst option. The vendored code will not be updated with new fuzzywuzzy code over time, and my goal is to eventually be able to remove the vendored code, as part of bskinn/sphobjinv#178 and following work.

File: recipes/sphobjinv/meta.yaml (hunk 3 above)
Comment (bskinn) [reply recipes/sphobjinv/meta.yaml:49]: Aha, nevermind -- this is a separate problem.

This should fix:

```suggestion
    - pytest -k "not readme"
```

Comment (bskinn) [reply recipes/sphobjinv/meta.yaml:49]: (depending on whether YAML needs special escaping for double-quotes...can't recall......)

**Comment by ocefpaf** [comment]
> but still test it across various OSes. @ocefpaf may be able to confirm this.

Nope. It would be nice but the whole idea of noarch is that the test is equivalent on the OSes and platforms. What you can do is to try the "brand new" some-arch packages. Those are packages that have platform specific stuff, build and test on each platform but not all all Pythons.

**Comment by bskinn** [comment]
I test `sphobjinv` aggressively cross-platform (Linux/MacOS/Win), cross-Python-version, and cross-dependency-version on all my releases, so I'm pretty comfortable with the conda-forge recipe only testing on POSIX.  The odds of a platform-specific bug emerging between my tests and the conda-forge build seem awfully small.

**Comment by ocefpaf** [comment]
> so I'm pretty comfortable with the conda-forge recipe only testing on POSIX

That is what I needed to ~~hear~~ (read).

**Pull request merged**